helper/augroups.nix

```diff
@@ -0,0 +1,189 @@
+          }
+        '';
+      };
+      vimCallback = mkOption {
+        default = null;
+        type = nullOr str;
+        description = ''
+          Vimscript function name called when the event(s) is triggered.
+
+          Conflicts with luaCallback
+        '';
+      };
+      command = mkOption {
+        default = null;
+        type = nullOr str;
+        description = ''
+          Vim command to execute on event. Cannot be used with {lua,vim}Callback
+        '';
+      };
+      once = mkOption {
+        default = false;
+        type = bool;
+        description = ''
+          Run the autocommand only once |autocmd-once|
+        '';
+      };
+      nested = mkOption {
+        default = false;
+        type = bool;
+        description = ''
+          Run nested autocommands |autocmd-nested|.
+        '';
+      };
+    };
+  };
+
+  quote = string: "\"${string}\"";
+
+  toLuaStringList = strings:"{${concatMapStringsSep "," quote strings}}";
+
+  genAutocmd = group: { event
+               , pattern
+               , buffer
+               , desc
+               , luaCallback
+               , vimCallback
+               , command
+               , once
+               , nested
+               }: let
+               # TODO: Assert that only one of luaCallback vimCallback or command is set
+               events = if builtins.isList event then toLuaStringList event else "{${quote event}}";
+               opts = {
+                 inherit group pattern buffer desc command once nested;
+                 callback = if luaCallback == null then
+                   vimCallback
+                 else
+                   rawLua ''
+                   function(opts)
+                     ${luaCallback}
+                   end
+                   '';
+               };
+             in ''
+    do
+      local events = ${events}
+      local opts = ${toLuaObject opts}
+      vim.api.nvim_create_autocmd(events, opts)
+    end
+  '';
+
+  genAugroup = {name, autocmds, clear}: let
+    opts = toLuaObject {inherit clear;};
+    group = rawLua "group";
+    autocmds' = concatMapStringsSep "\n" (genAutocmd group) autocmds;
+  in ''
+    do
+      local group = vim.api.nvim_create_augroup(${quote name}, ${opts})
+      ${autocmds'}
+    end
+  '';
+
+  genAugroups = augroups:
+    let
+      setName = name: opts: {inherit name;} // opts;
+      augroups' = mapAttrsToList setName augroups;
+    in
+      concatMapStringsSep "\n" genAugroup augroups';
+in {
+  augroupOptions = submodule {
+    options = {
+      name = mkOption {
+        type = str;
+        description = ''
+          The name of the augroup. If undefined, the name of the attribute set will be used.
+        '';
+      };
+      autocmds = mkOption {
+        type = listOf (submodule autocmdOpts);
+        description = ''
+          The autocmds that are part of this augroup.
+
+          See :help nvim_create_autocmd()
+        '';
+      };
+      clear = mkOption {
```

Comment:
In `helper/custom_options.nix` I have defined several helper functions, which make writing simple bool options much shorter. Could you make use of them here?